pool/pool.gno

```diff
@@ -498,17 +656,12 @@ func SetFeeProtocol(
 	)
 }
 
-func setFeeProtocol(
-	feeProtocol0 uint8,
-	feeProtocol1 uint8,
-) uint8 {
+func setFeeProtocol(feeProtocol0, feeProtocol1 uint8) uint8 {
 	common.IsHalted()
 
-	fee0Cond := feeProtocol0 == 0 || (feeProtocol0 >= 4 && feeProtocol0 <= 10)
-	fee1Cond := feeProtocol1 == 0 || (feeProtocol1 >= 4 && feeProtocol1 <= 10)
-	if !(fee0Cond && fee1Cond) {
+	if err := validateFeeProtocol(feeProtocol0, feeProtocol1); err != nil {
 		panic(addDetailToError(
-			errInvalidProtocolFeePct,
+			err,
```

Comment:
Since the `validateFeeProtocol` function already returns the `errInvalidProtocolFeePct` error code. So, it's sufficient to throw the returned error here.